app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/HistoryListAdapter.java

```diff
@@ -88,17 +88,35 @@ public void calcTime(Date date, HistoryScanHolder holder) {
 
 
         String secText = String.valueOf(seconds) + " seconds ago";
+
         String minText = String.valueOf(minutes) + " minutes ago";
+        String OneMinText = String.valueOf(minutes) + " minute ago";
+
         String hourText = String.valueOf(hours) + " hours ago";
+        String OneHourText = String.valueOf(hours) + " hour ago";
+
         String dayText = String.valueOf(days) + " days ago";
+        String OneDayText = String.valueOf(days) + " day ago";
 
         if (seconds < 60) {
             holder.txtDate.setText(secText);
-        } else if (minutes < 60) {
+        }
+        else if(minutes==1){
```

Comment:
(minutes == 1 ) condition will overlap with the condition (minutes <60) as one minute is less than 60 minutes .

Reply:
The same with hour and day.

Reply:
Try to add the (minute == 1) condition inside (minutes<60)condition . This will prevent overlap and make (minute == 1) a subset of (minutes<60) condition.

Reply:
@PrajwalM2212 Please see my above requested changes, this will not need to be done once my changes are implemented.

Reply:
@Karljoones  Yes . I will delete this comment .

Reply:
Hey @PrajwalM2212 , In my Humble Opinion the control flow for how else if conditions are designed inherently will take care of the specific flaw you are pointing out. Thanks, I'm already  working on @Karljoones  changes.

Reply:
@ManavPoddar  I know . I pointed out the issue to make the code look good and easily understandable for future developers.